File: Difficulty: Easy/Largest Pair Sum/largest-pair-sum.cpp
//{ Driver Code Starts
// Initial template for C++
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends

class Solution {
  public:
    int pairsum(vector<int> &arr) {
        // code here
        int n=arr.size();
        int max1=INT_MIN;
        int max2=INT_MIN;
        // for finding 1 largest
        for(int i=0;i<n;i++){
            if(arr[i]>max1){
                max1=arr[i];
            }
        }
        // for finding 2nd largest
         for(int i=0;i<n;i++){
            if(arr[i]>max2 && arr[i]!=max1){
                max2=arr[i];
            }
        }
        // returning the sum of 1st and 2nd largest 
        return max1+max2;
    }
};


//{ Driver Code Starts.
int main() {
    int t;

    cin >> t;
    cin.ignore();
    while (t--) {
        vector<int> arr;
        string input;
        getline(cin, input);
        stringstream ss(input);
        int number;
        while (ss >> number) {
            arr.push_back(number);
        }
        int n = arr.size();
        Solution ob;
        int r = ob.pairsum(arr);
        cout << r << endl;
    }
    return 0;
}
// } Driver Code Ends

File: Difficulty: Easy/Largest Pair Sum/largest-pair-sum.py

from typing import List


class Solution:
    def pairsum(self, arr : List[int]) -> int:
        # code here
        arr.sort()
        n=len(arr)
        i=n-1
        maxi=arr[i]
        while i:
            if maxi!=arr[i]:
                return maxi+arr[i]
            i-=1
        
        return maxi
        



#{ 
 # Driver Code Starts
class IntArray:

    def __init__(self) -> None:
        pass

    def Input(self, n):
        arr = [int(i) for i in input().strip().split()]  #array input
        return arr

    def Print(self, arr):
        for i in arr:
            print(i, end=" ")
        print()


if __name__ == "__main__":
    t = int(input())
    for _ in range(t):

        arr = list(map(int, input().strip().split()))

        obj = Solution()
        res = obj.pairsum(arr)

        print(res)

# } Driver Code Ends